storage/s3.go

```diff
@@ -177,6 +179,38 @@ func (s *S3) Delete(ctx context.Context, path string) error {
 	return nil
 }
 
+// Info returns several information contained in the header.
+// It returns ObjectNotFound custom error if the object does not exists.
+func (s *S3) Info(ctx context.Context, path string) (types.Info, error) {
+	path = fullPath(path)
+	var res *s3.HeadObjectOutput
+	err := s.retryWrapper(ctx, InfoMethod, func(ctx context.Context) error {
+		input := &s3.HeadObjectInput{Bucket: &s.cfg.Bucket, Key: &path}
+		var err error
+		res, err = s.s3client.HeadObject(ctx, input)
+		if err != nil {
+			return err
+		}
+		return nil
+	})
+
+	if err != nil {
+		var apiErr smithy.APIError
+		if stderrors.As(err, &apiErr) {
+			if apiErr.ErrorCode() == NotFoundErrCode {
+				return types.Info{}, ObjectNotFound{}
+			}
+		}
+		return types.Info{}, errors.Wrapf(err, "fail to HEAD object '%v'", path)
+	}
+
+	return types.Info{
+		ContentLength: res.ContentLength,
+		ContentType:   *res.ContentType,
+		Checksum:      *res.ChecksumSHA256,
```

Comment:
Please add the type of CHecksum too ;-) ChecksumSHA256, it may evolve over time.

THanks!